cms/static/cms/js/modules/cms.toolbar.js

```diff
@@ -17,106 +32,88 @@
                 }
 
-                // check if debug is true
-                if (CMS.config.debug) {
-                    this._debug();
-                }
-
-                // check if there are messages and display them
-                if (CMS.config.messages) {
-                    this.openMessage(CMS.config.messages);
-                }
-
-                // check if there are error messages and display them
-                if (CMS.config.error) {
-                    this.showError(CMS.config.error);
-                }
-
-                // enforce open state if user is not logged in but requests the toolbar
-                if (!CMS.config.auth || CMS.config.settings.version !== this.settings.version) {
-                    this.toggleToolbar(true);
-                    this.settings = this.setSettings(CMS.config.settings);
-                }
-
-                // should switcher indicate that there is an unpublished page?
-                if (CMS.config.publisher) {
-                    this.openMessage(CMS.config.publisher, 'right');
-                    setInterval(function () {
-                        CMS.$('.cms-toolbar-item-switch').toggleClass('cms-toolbar-item-switch-highlight');
-                    }, this.options.messageDelay);
-                }
-
-                // open sideframe if it was previously opened
-                if (this.settings.sideframe.url) {
-                    var sideframe = new CMS.Sideframe();
-                    sideframe.open(this.settings.sideframe.url, false);
-                }
+                // FIXME the general initialization is handled within the toolbar
+                // rather than a separate cms.setup or similar. Yet other components
+                // are loaded after the toolbar so it can create a clash where
+                // CMS.API is not ready. This is a workaround until a proper fix
+                // will be released in 3.x
+                var that = this;
+                setTimeout(function () {
+                    that._initialStates();
+                }, 200);
 
-                // if there is a screenblock, do some resize magic
-                if (this.screenBlock.length) {
-                    this._screenBlock();
-                }
+                // set a state to determine if we need to reinitialize this._events();
+                this.ui.toolbar.data('ready', true);
+            },
 
-                // add toolbar ready class to body and fire event
-                this.body.addClass('cms-ready');
-                $(document).trigger('cms-ready');
+            /**
+             * Stores all jQuery references within `this.ui`.
+             *
+             * @method _setupUI
+             * @private
+             */
+            _setupUI: function _setupUI() {
+                var container = $('#cms-toolbar');
+                this.ui = {
+                    container: container,
+                    body: $('html'),
```

Comment:
can we change the name to something else (similar to what's in modal)? body is misleading here